cmake/WindowsEncoding.cmake

```diff
@@ -0,0 +1,4 @@
+if(CMAKE_C_COMPILER_ID STREQUAL "MSVC" OR CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  set(CMAKE_C_FLAGS   "${CMAKE_C_FLAGS}   /source-charset:utf-8 /execution-charset:gbk")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /source-charset:utf-8 /execution-charset:gbk")
+endif()
```

Comment:
I'm not really in favor of setting the `execution-charset`, I think it should not be necessary, in https://learn.microsoft.com/en-us/cpp/build/reference/execution-charset-set-execution-character-set?view=msvc-170 I see:
> By default, Visual Studio detects a byte-order mark to determine if the source file is in an encoded Unicode format, for example, UTF-16 or UTF-8. If no byte-order mark is found, it assumes that the source file is encoded in the current user code page, unless you used the /source-charset or /utf-8 option to specify a character set name or code page. 

So it is looks like a local problem of the "wrong" user code page but it should be sufficient to set the `source-charset` (maybe together with `/execution-charset:utf-8`).

Reply:
Tried as your advice, only keep `/source-charset:utf-8` can also succesfully compile with VS2022.

I was not very clear about `/execution-charset` option, due to not necessary setting it, removed now.